editor.planx.uk/src/@planx/components/Calculate/Editor.tsx

```diff
@@ -36,12 +33,49 @@ export default function Component(props: Props) {
         props.handleSubmit({ type: TYPES.Calculate, data: newValues });
       }
     },
-    validate: () => {
-      // can parse formula
-      getVariables(formik.values.formula);
+    validate: (values) => {
+      const errors: FormikErrors<Calculate> = {};
+      try {
+        // can parse formula
+        getVariables(formik.values.formula);
+
+        // Validate formula
+        const result = evaluate(
+          values.formula,
+          values.samples,
+          values.defaults,
+        );
+
+        if (Number.isNaN(Number(result))) {
+          errors.formula = "Enter a formula which outputs a number";
+        }
+      } catch (error: any) {
+        errors.formula = error.message;
+      }
+      return errors;
     },
+
+    validateOnChange: false,
   });
 
+  const sampleResult = React.useMemo(() => {
+    try {
+      const result = evaluate(
+        formik.values.formula,
+        formik.values.samples,
+        formik.values.defaults,
+      );
+      // Type guard as mathjs evaluates `m` to a "Unit" object for "meter"
+      if (!Number.isNaN(Number(result))) {
+        return result;
+      } else {
+        return `the ${typeof result}: '${result}'`;
```

Comment:
Added this to ensure clear error messaging is given to Editors

Reply:
![image](https://github.com/user-attachments/assets/144c6ba7-e035-494f-8630-f77b84be3809)

I think it's worth looking at how this is handling undefined values.

Reply:
Now got it showing nothing if undefined, which feels right. I kept getting weird behaviour on existing Calculate components when using `formik.dirty` so I've not used it for conditional rendering